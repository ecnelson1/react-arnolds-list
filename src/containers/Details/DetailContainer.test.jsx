import React from 'react';
import {render, screen} from '@testing-library/react';
import {rest} from 'msw';
import {setupServer} from 'msw/node';
import DetailsContainer from './DetailContainer';
import mockData from  '../../../fixtures/mock-data.json';

const server = setupServer(
    rest.get('https://hey-arnold-api.herokuapp.com/api/v1/characters', (req, res, ctx) => {
        return res(ctx.json(mockData));
    })
);

describe('DetailsContainer', () => {
    beforeAll(() => server.listen());
    afterAll(() => server.close());
    
    it('renders a loading element', async () => {
        render(<DetailsContainer match={{params: {id:'5da237699734fdcb7bef8f52'}}}/>);
        screen.getByText('LOADING...');
    });

    it('renders a character from Hey Arnold by matching id', async ()=> {
        render(<DetailsContainer match={{params: {id:'5da237699734fdcb7bef8f52'}}}/>);
        const figure = await screen.findByRole('figure', { name: 'character'});
        expect(figure).toMatchSnapshot();
    });
});
