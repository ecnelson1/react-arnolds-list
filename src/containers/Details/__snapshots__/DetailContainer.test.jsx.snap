// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsContainer renders a character from Hey Arnold by matching id 1`] = `
<figure
  aria-label="character"
>
  <img
    alt="Grandpa Phil"
    src="https://vignette.wikia.nocookie.net/heyarnold/images/7/79/GrandpaPhil.png/revision/latest/scale-to-width-down/300?cb=20160924030436"
  />
  <figcaption>
    Grandpa Phil
  </figcaption>
</figure>
`;
